creator-node/src/routes/tracks.js

```diff
@@ -50,29 +56,122 @@ module.exports = function (app) {
     syncLockMiddleware,
     handleTrackContentUpload,
     handleResponse(async (req, res) => {
-      const AsyncProcessingQueue =
-        req.app.get('serviceRegistry').asyncProcessingQueue
-
       if (req.fileSizeError || req.fileFilterError) {
         removeTrackFolder({ logContext: req.logContext }, req.fileDir)
         return errorResponseBadRequest(req.fileSizeError || req.fileFilterError)
       }
 
-      await AsyncProcessingQueue.addTrackContentUploadTask({
+      const AsyncProcessingQueue =
+        req.app.get('serviceRegistry').asyncProcessingQueue
+
+      const currentNodeCanHandleTranscode = canCurrentNodeHandleTranscode({
+        transcodingQueueCanAcceptMoreJobs: await TranscodingQueue.isAvailable(),
+        spID: config.get('spID'),
+        libs: AsyncProcessingQueue.getLibs()
+      })
+
+      if (currentNodeCanHandleTranscode) {
+        await AsyncProcessingQueue.addTrackContentUploadTask({
+          logContext: req.logContext,
+          req: {
+            fileName: req.fileName,
+            fileDir: req.fileDir,
+            fileDestination: req.file.destination,
+            cnodeUserUUID: req.session.cnodeUserUUID
+          }
+        })
+      } else {
+        await AsyncProcessingQueue.addTranscodeHandOffTask({
+          logContext: req.logContext,
+          req: {
+            fileName: req.fileName,
+            fileDir: req.fileDir,
+            fileNameNoExtension: req.fileNameNoExtension,
+            fileDestination: req.file.destination,
+            cnodeUserUUID: req.session.cnodeUserUUID,
+            headers: req.headers
+          }
+        })
+      }
+
+      return successResponse({ uuid: req.logContext.requestID })
+    })
+  )
+
+  /**
+   * Given that the requester is a valid SP, the current Content Node has enough storage,
+   * upload the track to the current node and add a transcode and segmenting job to the queue.
+   *
+   * This route is part of the transcode handoff logic.
+   */
+  app.post(
+    '/transcode_and_segment',
+    ensureValidSPMiddleware,
+    ensureStorageMiddleware,
+    handleTrackContentUpload,
+    handleResponse(async (req, res) => {
+      const AsyncProcessingQueue =
+        req.app.get('serviceRegistry').asyncProcessingQueue
+
+      await AsyncProcessingQueue.addTranscodeAndSegmentTask({
         logContext: req.logContext,
         req: {
           fileName: req.fileName,
           fileDir: req.fileDir,
-          fileDestination: req.file.destination,
-          session: {
-            cnodeUserUUID: req.session.cnodeUserUUID
-          }
+          uuid: req.logContext.requestID
         }
       })
+
       return successResponse({ uuid: req.logContext.requestID })
     })
   )
 
+  /**
+   * Given that the request is coming from a valid SP, serve the corresponding file
+   * from the transcode handoff
+   */
+  app.get(
+    '/transcode_and_segment',
```

Comment:
would it be clearer as `app.get('/transcoded_segments')`? or do we need to keep the same route name for the way the logic is handled around retries?

Reply:
I feel like tracksSecondary.js is kind of a confusing name and maybe we should keep everything in tracks.js if it's related to tracks

Reply:
right now it's not clear what happens on the primary vs handoff node. at the very least we should make the comment a little bit clearer like 
```
This does not run on the user's primary. It's sent from the primary to the current node to handoff track transcoding.
```

Reply:
added comments to both the post/get routes